python/ray/external_storage.py

```diff
@@ -90,6 +103,119 @@ def restore_spilled_objects(self, keys):
                 # read remaining data to our buffer
                 self._put_object_to_store(metadata, buf_len, f, ref)
 
+    def validate(self):
+        if not os.path.exists(self.directory_path):
+            raise ValueError("The give directory path to store objects, "
+                             f"{self.directory_path}, doesn't exist.")
+
+
+class S3Storage(ExternalStorage):
```

Comment:
Split into a separate file?

Reply:
> It doesn't modularize different adapters as well as defines proper interfaces. This also needs more discussion (for the right "interface").

Will have a separate PR to refactor (and would like to get your sign off before I make that PR to avoid disagreement)

Reply:
Sounds good.